python/lsst/daf/butler/datastores/webdavDatastore.py

```diff
@@ -181,6 +181,9 @@ def _write_in_memory_to_artifact(self, inMemoryDataset: Any, ref: DatasetRef) ->
             raise FileExistsError(f"Cannot write file for ref {ref} as "
                                   f"output file {location.uri} exists.")
 
+        if not location.uri.dirname().exists():
```

Comment:
Need to consider how we deal with this in the new RemoteFileDatastore that has replaced this implementation. For now we should probably just add it and absorb the one extra S3 call.